src/internal/ajax/ajax.ts

```diff
@@ -295,6 +295,16 @@ export function fromAjax<T>(config: AjaxConfig): Observable<AjaxResponse<T>> {
       headers['x-requested-with'] = 'XMLHttpRequest';
     }
 
+    // Allow users to provide their XSRF cookie name and the name of a custom header to use to
+    // send the cookie.
+    const { withCredentials, xsrfCookieName, xsrfHeaderName } = config;
+    if ((withCredentials || !config.crossDomain) && xsrfCookieName && xsrfHeaderName) {
+      const xsrfCookie = readCookie(xsrfCookieName);
```

Comment:
I think we can inline the entire `readCookie` call here to save on size. And since we'd just be rethrowing the error anyhow, that means this would just be:

```ts
const { withCredentials, xsrfCookieName, crossDomain, xsrfHeaderName } = config;
((withCredentials || !crossDomain) && xsrfCookieName && xsrfHeaderName) && (headers[xsrfHeaderName] = document?.cookie?.match(new RegExp(`(^|;\\s*)(${name})=([^;]*)`))?.pop() ?? '');
```

I know that *looks* ridiculous, but we're trying to be size-sensitive with our library because so many people use it.

Reply:
Done.